### PR TITLE
Drop Discovery, drop mailings, fix pipeline

### PR DESCRIPTION
**Pull request by fghaas**
As reported by @dyudyunov in #39, the Ecommerce service isn't the only one for which retirement is breaking.

Thus:

* Drop support for the Discovery service.
* Drop support for the mailing API that the LMS dropped in Ironwood.
* Fix the order of pipeline invocation.

These changes are fixed up from https://github.com/GSVlabs/tutor-contrib-philu-retirement, with added Changelog and README updates.

Fixes: #39

### CODE REVIEW

**Review by mrtmm — APPROVED** [review]

**Comment by fghaas** [comment]
@dyudyunov, do you have any second thoughts, or objections to us merging this?

File: README.md [hunk 1]
```diff
@@ -31,9 +31,10 @@ appropriate one:
 Limitations
 ------------
 
-This plugin cannot be used for retiring accounts in the [Open edX E-Commerce Service](https://github.com/openedx/ecommerce).[^ecom]
+This plugin cannot be used for retiring accounts in the [Open edX E-Commerce Service](https://github.com/openedx/ecommerce),[^ecom] nor the [Course Discovery Service](https://github.com/openedx/course-discovery).[^discovery].
 
-[^ecom]: See [Issue #36](https://github.com/hastexo/tutor-contrib-retirement/issues/36) in this repository for a related discussion of missing functionality in E-Commerce account retirement.
+[^ecom]: See [Issue #36](https://github.com/hastexo/tutor-contrib-retirement/issues/36) for background.
+[^discovery]: See [Issue #37](https://github.com/hastexo/tutor-contrib-retirement/issues/36) for background.
```
Comment (dyudyunov) [inline README.md:37]: The link possibly has a typo: Discovery issue is described in issue #39, not in the #36 (the e-commerce issue).

And the link name is `Issue #37` but it leads to #36 if I understand it correctly.

```suggestion
[^discovery]: See [Issue #39](https://github.com/hastexo/tutor-contrib-retirement/issues/39) for background.
```

**Review by dyudyunov — APPROVED** [review]
@fghaas I have one suggestion, please check

Thanks for the PR! 👍

File: README.md (hunk 1 above)
Comment (fghaas) [reply README.md]: Ah yes, good catch, thanks! Fixed in 2d3d23399f126b3c0cf17b1c1102787a456bbcc6.

**Review by mrtmm — APPROVED** [review]

**Pull request merged**